presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/DefaultCachingJdbcClientFactory.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.jdbc;
+
+import io.airlift.units.Duration;
+
+import javax.inject.Inject;
+
+import java.util.concurrent.TimeUnit;
+
+import static java.util.Objects.requireNonNull;
+
+public class DefaultCachingJdbcClientFactory
+        implements CachingJdbcClientFactory
+{
+    private final JdbcClient jdbcClient;
+
+    @Inject
+    public DefaultCachingJdbcClientFactory(JdbcClient jdbcClient)
+    {
+        this.jdbcClient = requireNonNull(jdbcClient, "jdbcClient is null");
+    }
+
+    @Override
+    public JdbcClient create()
+    {
+        return new CachingJdbcClient(jdbcClient, new Duration(1, TimeUnit.DAYS), true);
```

Comment:
Where does 1d come from?
sounds like "almost infinity"

use io.prestosql.plugin.jdbc.BaseJdbcConfig#getMetadataCacheTtl here

Reply:
> sounds like "almost infinity"

That makes sense within single transaction. The value was moved from previous code version (see `JdbcMetadataFactory`)

Reply:
1d is neither "reasonably short for something" not "infinite"
I am OK not changing this, but maybe "TODO use infinity for in-transaction cache"

Reply:
> I am OK not changing this, but maybe "TODO use infinity for in-transaction cache"

I don't think planner will ask for stats after one day ;)

Reply:
this is about any caching, not related to stats

if i have a very long running transaction, it does not make sense to flush state after 24h.
if we want to limit trx length, we should do this explciitly (& somewhere else)